src/nvim/globals.h

```diff
@@ -72,7 +72,7 @@ EXTERN long Rows                        /* nr of rows in the screen */
   = 24L
 #endif
 ;
-EXTERN long Columns INIT(= 80);         /* nr of columns in the screen */
+EXTERN int Columns INIT(= 80);         /* nr of columns in the screen */
```

Comment:
There are about 240 usages of global `Columns`. Have you checked all of them?
Notice that the fact that it compiles without warnings after this change doesn't mean anything, as many of those usages can be in files where -Wconversion is not active yet. And, in files where -Wconversion is enabled, casts (and corresponding previous checks through asserts) may be present that could be removed.
So, to do it right, you have to manually check each occurrence. That's why this has been proceeding slowly. It's not so easy as it can seem. In fact, I don't consider enabling -Wconversion an entry-level task.